### PR TITLE
An initial use-cases document for the web packaging proposal.

### PR DESCRIPTION
**Pull request by jyasskin**
Preview at https://jyasskin.github.io/webpackage/draft-webpackage-use-cases/draft-yasskin-webpackage-use-cases.html

@mnot, you seemed to have some use cases for signing individual request/response/body triples, but I don't remember what they were. Could you list a couple so I can add them to this document?

### CODE REVIEW

File: draft-yasskin-webpackage-use-cases.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+The package's length in bytes appears a fixed offset from the end of
+the package.
+
+# Non-goals
+
+Some features often come along with packaging and signing, and it's
+important to explicitly note that they don't appear in the list of
+{{requirements}}{:format="title"}.
+
+## Store confidential data {#confidential}
```
Comment (mrdewitt) [inline draft-yasskin-webpackage-use-cases.md]: Is this worth a little more enumeration?  Confidentiality comes up repeatedly in discussions and it'd be good to ensure that rehashing is minimized.

File: draft-yasskin-webpackage-use-cases.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+network. The W3C has addressed running applications offline with
+Service Workers ({{ServiceWorkers}}), but not the problem of
+distribution.
+
+
+# Use cases
+
+These use cases are in rough descending priority order. If use cases
+have conflicting requirements, the design should enable more important
+use cases.
```
Comment (mrdewitt) [inline draft-yasskin-webpackage-use-cases.md]: Is there some line in the list of priorities that we won't cross?  Is Http/2 push essential to this document? is offline browsing?

Comment (jyasskin) [inline draft-yasskin-webpackage-use-cases.md]: I'd say that everything below https://jyasskin.github.io/webpackage/draft-webpackage-use-cases/draft-yasskin-webpackage-use-cases.html#security-review is nice-to-have. Do you think I should split the use-cases into "essential" and "nice-to-have" buckets to reflect that?

File: draft-yasskin-webpackage-use-cases.md (hunk 1 above)
Comment (jyasskin) [inline draft-yasskin-webpackage-use-cases.md]: How's this?

File: draft-yasskin-webpackage-use-cases.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+
+## Offline installation {#offline-installation}
+
+Alex can download a file containing a website
+(a [PWA](https://developers.google.com/web/progressive-web-apps/checklist))
+including a Service Worker from origin `O`, and transmit it to their peer Bailey,
+and then Bailey can install the Service Worker with a proof that it came from `O`.
+This saves Bailey the bandwidth costs of transferring the website.
+
+Associated requirements:
```
Comment (jyasskin) [inline draft-yasskin-webpackage-use-cases.md]: I'll add a sentence to all the requirements. That's not done yet.

Comment (jyasskin) [inline draft-yasskin-webpackage-use-cases.md]: Done.

File: draft-yasskin-webpackage-use-cases.md (hunk 2 above)
Comment (mrdewitt) [inline draft-yasskin-webpackage-use-cases.md]: Thanks!

File: draft-yasskin-webpackage-use-cases.md (hunk 1 above)
Comment (mrdewitt) [inline draft-yasskin-webpackage-use-cases.md]: thanks, looks good

**Pull request merged**